dozer-api/src/rest/tests/routes.rs

```diff
@@ -63,24 +66,37 @@ where
     if let Some(query) = query.clone() {
         req = req.set_json(query);
     }
+    if content_type.is_some() {
+        req = req.insert_header(content_type.clone().unwrap());
+    }
     let req = req.to_request();
     let res = actix_web::test::call_service(service, req).await;
-    assert!(res.status().is_success());
-
-    let body: Value = actix_web::test::read_body_json(res).await;
-    let count = body.as_u64().unwrap();
+    let mut count = 0;
+    let mut records: Vec<Value> = vec![];
+    if content_type.is_some() && content_type.clone().unwrap() != ContentType::json() {
+        assert_eq!(res.status(), StatusCode::BAD_REQUEST);
+    } else {
+        assert!(res.status().is_success());
+        let body: Value = actix_web::test::read_body_json(res).await;
+        count = body.as_u64().unwrap();
+    }
```

Comment:
```suggestion
    res.error_for_status()?;
    let body: Value = actix_web::test::read_body_json(res).await;
    let count = body.as_u64().unwrap();
```

Reply:
`error_for_status` is not present in `actix-web`. Used `res.status().is_success()` to check, and manually return the corresponding status code.